featuregate/flag_test.go

```diff
@@ -14,97 +14,151 @@ func TestNewFlag(t *testing.T) {
 	for _, tt := range []struct {
```

Comment:
Can we have a test where we test that both 
```
--feature-gates=alpha,beta --feature-gates-strict
```

and

```
--feature-gates-strict --feature-gates=alpha,beta
```

work?

Reply:
Sure thing. I tried manually and it worked, maybe we need an integration test for this.

Reply:
> Sure thing. I tried manually and it worked, maybe we need an integration test for this.

I think we can make it work with a unit test: for each `arguments` combination
1. call the `flags` function with a new registry,
2. call [`flagset.Parse`](https://pkg.go.dev/flag#FlagSet.Parse) with `arguments`
3. check that the error is as expected, or that the registry has the values you expect otherwise.

That should do the trick

Reply:
ok I will do that for sure, thanks. Sorry, I'll get it done soon, juggling work.

Reply:
No problem :) not my intention to introduce any sense of urgency, sorry if it came out that way

Reply:
No, just feeling mad at myself for not getting this right :)

Reply:
Took me a good while but I think I figured it out. There was indeed an ordering issue.